lib/IDE/SyntaxModel.cpp

```diff
@@ -418,6 +418,21 @@ CharSourceRange parameterNameRangeOfCallArg(const TupleExpr *TE,
   return CharSourceRange();
 }
 
+static CharSourceRange findDocCommentRange(DeclAttributes Attrs) {
+  for (auto Attr : Attrs) {
+    if (auto DocAttr = dyn_cast_or_null<RawDocCommentAttr>(Attr)) {
+      return DocAttr->getCommentRange();
```

Comment:
Could we add a function `getCharSourceRange()` to RawComment that analyses the enclosed `SingleRawComment` where range is available?